src/modules/kvs/kvs.c

```diff
@@ -1350,65 +1390,23 @@ static void setroot_event_cb (flux_t *h, flux_msg_handler_t *w,
     kvs_ctx_t *ctx = arg;
     int rootseq;
     const char *rootref;
-    json_t *root = NULL;
+    json_t *rootdir = NULL;
     json_t *names = NULL;
 
     if (flux_event_unpack (msg, NULL, "{ s:i s:s s:o s:o }",
                            "rootseq", &rootseq,
                            "rootref", &rootref,
                            "names", &names,
-                           "rootdir", &root) < 0) {
+                           "rootdir", &rootdir) < 0) {
         flux_log_error (ctx->h, "%s: flux_event_unpack", __FUNCTION__);
         return;
     }
 
     finalize_fences_bynames (ctx, names, 0);
-    /* Copy of root object (corresponding to rootref) was included
-     * in the setroot event as an optimization, since it would otherwise
-     * be loaded from the content store on next KVS access - immediate
-     * if there are watchers.  Store this object in the KVS cache
-     * with clear dirty bit as it is already valid in the content store.
-     */
-    if (!json_is_null (root)) {
-        struct cache_entry *hp;
-        if ((hp = cache_lookup (ctx->cache, rootref, ctx->epoch))) {
-            if (!cache_entry_get_valid (hp)) {
-                /* On error, bad that we can't cache new root, but
-                 * no consistency issue by not caching.  We will still
-                 * set new root below via setroot().
-                 */
-                if (cache_entry_set_treeobj (hp, root) < 0) {
-                    flux_log_error (ctx->h, "%s: cache_entry_set_treeobj",
-                                    __FUNCTION__);
-                }
-            }
-            if (cache_entry_get_dirty (hp)) {
-                /* If it was dirty, an RPC is already in process, so
-                 * let that RPC handle any error handling with this
-                 * cache entry, we just log this error.
-                 */
-                if (cache_entry_set_dirty (hp, false) < 0)
-                    flux_log_error (ctx->h, "%s: cache_entry_set_dirty",
-                                    __FUNCTION__);
-            }
-        } else {
-            /* On error, bad that we can't cache new root, but
-             * no consistency issue by not caching.  We will still
-             * set new root below via setroot().
-             */
-            if (!(hp = cache_entry_create ())) {
-                flux_log_error (ctx->h, "%s: cache_entry_create",
-                                __FUNCTION__);
-            }
-            else if (cache_entry_set_treeobj (hp, root) < 0) {
-                flux_log_error (ctx->h, "%s: cache_entry_set_treeobj",
-                                __FUNCTION__);
-                cache_entry_destroy (hp);
-            }
-            else
-                cache_insert (ctx->cache, rootref, hp);
-        }
-    }
+
+    if (!json_is_null (rootdir))
```

Comment:
i think it's important to leave a comment here indicating that if an error occurs in ```prime_cache_with_rootdir()```, it's no big deal.  We still set the setroot and things are generally fine.

Reply:
I did make the function return void and put a note about failure being OK over the function.  More blatant warning needed you think?

Reply:
yeah, just a bit more blatant is what I'm thinking.